### PR TITLE
tests: samples: applications: Replacement native_posix to native_sim

### PR DESCRIPTION
**Pull request by katgiadla**
Zephyr RTOS recommends using `native_sim` platform as default simulator.
Native simulator provides a few peripheral to extend possibilites to developing and tests.

These changes replace and customize files to using `native_sim` at sdk-nrf,

Refers: NCSDK-28086

### CODE REVIEW

**Review by greg-fer — APPROVED** [review]
Does this warrant an entry in changelog? Your call, as always for changelog entries.

File: scripts/west_commands/pygit2_helpers.py
Comment (PerMac) [inline scripts/west_commands/pygit2_helpers.py:233]: I don't think this needs a change. @aescolar?

**Review by PerMac — CHANGES_REQUESTED** [review]

File: scripts/quarantine_windows_mac.yaml
Comment (aescolar) [inline scripts/quarantine_windows_mac.yaml:11]: Beware that native_posix still exists as a platform in Zephyr. Though it is not anymore a default test platform. 
So this change may or may not be a problem depending on how twister is called (note that leaving native_posix as meant to be excluded should not be a problem)

**Review by SeppoTakalo — APPROVED** [review]

File: scripts/west_commands/pygit2_helpers.py
Comment (aescolar) [reply scripts/west_commands/pygit2_helpers.py:233]: This is the first time I see this west command, and I do not even know what it is meant to do. So I can't really help here :(

File: tests/subsys/suit/cache/testcase.yaml
Comment (aescolar) [inline tests/subsys/suit/cache/testcase.yaml:3]: note that there is also a native_sim/native/64
Which depending on wanting to test 64bit configuration may still make sense in the platform_allow

**Review by aescolar — COMMENTED** [review]
Thanks @katgiadla 
Just some minor concerns, which may be something or not depending on how these tests and code are used.

Note that the kconfig changes are effectively dropping support for native_posix (That may surprise some people)

File: subsys/net/lib/mqtt_helper/Kconfig
Comment (eivindj-nordic) [inline subsys/net/lib/mqtt_helper/Kconfig:88]: I think updating our tests to use `native_sim` is a good improvement. Though I wonder if we should continue to support `native_posix` as a board in NCS, at least for now. Customers might have tests running with `native_posix` board for their applications. Do we want to force them to change to `native_sim` as well?

```suggestion
	depends on (BOARD_QEMU_X86 || BOARD_NATIVE_SIM || BOARD_NATIVE_POSIX || BOARD_NRF7002DK_NRF5340_CPUAPP || BOARD_NRF7002DK_NRF5340_CPUAPP_NS) && MQTT_LIB_TLS
```

**Review by plskeggs — APPROVED** [review]

File: tests/subsys/suit/cache/testcase.yaml
Comment (katgiadla) [reply tests/subsys/suit/cache/testcase.yaml]: done

File: subsys/net/lib/mqtt_helper/Kconfig
Comment (carlescufi) [reply subsys/net/lib/mqtt_helper/Kconfig:88]: @aescolar could you weigh in please?

Comment (aescolar) [reply subsys/net/lib/mqtt_helper/Kconfig:88]: Apart from https://github.com/nrfconnect/sdk-nrf/pull/16584#discussion_r1688122345 ;)
It is up to somebody else to decide.
Anyhow, you can expect that in ~2 weeks I will deprecate native_posix in upstream Zephyr. So by the next "upmerge" the deprecation will be propagated to ncs.

**Review by tony-le-24 — APPROVED** [review]

**Comment by trantanen** [comment]
Is there a plan to get this PR forward?

File: tests/subsys/suit/cache_sink/src/main.c
Comment (jorgenmk) [inline tests/subsys/suit/cache_sink/src/main.c:15]: Fixing the git conflict markers in this file should get this PR through the twister stage

**Review by jorgenmk — CHANGES_REQUESTED** [review]

File: tests/subsys/suit/cache_sink/src/main.c
Comment (katgiadla) [reply tests/subsys/suit/cache_sink/src/main.c]: done. thanks :)

**Review by trantanen — APPROVED** [review]

**Review by juhaylinen — APPROVED** [review]

File: scripts/west_commands/pygit2_helpers.py
Comment (tejlmand) [reply scripts/west_commands/pygit2_helpers.py]: perhaps it should be added with `native_sim` in addition, ref: https://github.com/nrfconnect/sdk-nrf/commit/fff95136065278015530dde074a4602b8ee1d30d

although i'm in doubt if anyone actually is using this anymore to track important upstream changes:
> I used to use this for making blog posts about what changed in Zephyr.
We want to start including significant Zephyr changes in the NCS
release notes as well, so it will hopefully be useful for splitting up
the work among maintainers during upmerges.

**Review by tejlmand — APPROVED** [review]
Besides the quarantine files which pointed out by @aescolar should probably still be allowed to also used `native_posix`, see https://github.com/nrfconnect/sdk-nrf/pull/16584/files#r1688122345, then rest of changes lgtm.

**Review by umapraseeda — APPROVED** [review]

**Review by aescolar — APPROVED** [review]
LGTM. Just a trivial suggestion

**Review by tomchy — APPROVED** [review]

File: scripts/quarantine.yaml
Comment (katgiadla) [reply scripts/quarantine.yaml]: Thanks for suggestion :) changes applied

**Review by PerMac — APPROVED** [review]

**Review by jorgenmk — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (greg-fer) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:55]:
```suggestion
* Updated various tests and samples to use Zephyr's :ref:`native_sim<zephyr:native_sim>` instead of Zephyr's :ref:`native_posix<zephyr:native_posix>` for :ref:`running_unit_tests`.
  This mirrors the deprecation of ``native_posix`` in Zephyr.
  Support for ``native_posix`` will be removed in Zephyr with the v4.2 release.
```
If Zephyr v4.2 needs to be upmerged into sdk-nrf for this change to take effect, then the last sentences should be:
```
  Support for ``native_posix`` will be removed with the Zephyr v4.2 release.
  In the |NCS|, it will be removed once Zephyr v4.2 is upmerged to sdk-nrf.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Please confirm with Vestavind what the plan is for integrating the change planned for Zephyr v4.2 into sdk-nrf (upmerge or other method), and choose the correct phrasing.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (aescolar) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:55]: I confirm (from Vestavind) that the plan is that once native_posix is removed from upstream Zephyr in 4.2, NCS's sdk-zephyr will get that change in the next upmerge.

**Comment by PerMac** [comment]
@shanthanordic Is 2.8 milestone required on this one (you set the milestone)? If so, it needs backport. IMO this can go just to main and be in place for 2.9.

File: samples/app_event_manager/sample.yaml
Comment (MarekPieta) [inline samples/app_event_manager/sample.yaml:60]: Commit message title: `nativ_sim` -> `native_sim`

File: scripts/quarantine_integration.yaml
Comment (MarekPieta) [inline scripts/quarantine_integration.yaml:1430]: btw. Do we need to update quarantines (apart from sdk-zephyr)? The `native_posix` board seems removed in scope of the same PR.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (MarekPieta) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:41]: `Updated tests and samples`? Seems that you updated all of them

**Review by MarekPieta — APPROVED** [review]
A few minor comments to consider. Looks good

**Review by oyvindronningstad — APPROVED** [review]

File: scripts/quarantine_integration.yaml
Comment (katgiadla) [reply scripts/quarantine_integration.yaml:1430]: `native_posix` will dropping after Zephyr v.4.2 (on sdk-nrf when it will be upmerged). Until then, `native_posix` might be run on CI (or another CI plans). This entry avoid running these tests.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (katgiadla) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:41]: imo, word `various` highlights that changes concern all areas of tests/samples. 
@greg-fer , what is your opinion?

File: samples/app_event_manager/sample.yaml
Comment (katgiadla) [reply samples/app_event_manager/sample.yaml:60]: done. Thank you for vigilance :)

**Review by Damian-Nordic — APPROVED** [review]

**Review by 57300 — APPROVED** [review]
`tests/subsys/sdfw_services` LGTM

**Review by rugeGerritsen — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Comment by robertstypa** [comment]
The test-sdk-dfu tests are in red due to Coverity High Impact issues found in unit tests. These issues can be ignored, and the PR can be merged.

**Pull request merged**
